### PR TITLE
[workload-api] Remove lockfile, unused dependencies, and simplify build script

### PR DESCRIPTION
**Pull request by onalante-msft**
These changes reduce compile time significantly.

Summary:
- Remove lockfile:
  Since workload-api is being used as a library, `Cargo.lock` is not necessary.
- Remove tokio and reqwest build dependencies:
  Simpler to just shell out to curl.

### CODE REVIEW

File: common/workload-api/build.rs [hunk 1]
```diff
@@ -10,18 +10,19 @@
...
+            "--status",
+            "--strict",
+            "checksums.txt",
+        ])
+        .status()
+        .expect("failed to compute sha256 checksum");
 
-#[tokio::main]
-async fn main() {
-    let response = reqwest::get("https://raw.githubusercontent.com/spiffe/go-spiffe/v1.1.0/proto/spiffe/workload/workload.proto".to_string())
```
Comment (huguesBouvier) [inline common/workload-api/build.rs:17]: You are not pulling the proto from github anymore? Why remove that part?

File: common/workload-api/src/lib.rs [hunk 2]
```diff
@@ -9,16 +9,17 @@
     clippy::similar_names,
     clippy::too_many_lines
 )]
-pub mod client_and_server;
+pub mod generated {
+    // NOTE: workload.proto is without a "package" directive, so the generated code is in "${OUT_DIR}/_.rs"
+    include!(concat!(env!("OUT_DIR"), "/_.rs"));
```
Comment (huguesBouvier) [inline common/workload-api/src/lib.rs:14]: I guess this gets generated while building?

**Review by huguesBouvier — COMMENTED** [review]
Code cov is failing, you shoud check what's happening

File: common/workload-api/build.rs (hunk 1 above)
Comment (onalante-msft) [reply common/workload-api/build.rs]: If we pull from GitHub, we should validate the contents. Or, we can remove the validation and have the .proto checked in. I can add the fetch back with the checksum included.

**Review by huguesBouvier — APPROVED** [review]

**Pull request merged**